rasa/core/agent.py

```diff
@@ -392,156 +312,110 @@ class Agent:
     def __init__(
         self,
         domain: Union[Text, Domain, None] = None,
-        policies: Union[PolicyEnsemble, List[Policy], None] = None,
-        interpreter: Optional[NaturalLanguageInterpreter] = None,
         generator: Union[EndpointConfig, NaturalLanguageGenerator, None] = None,
         tracker_store: Optional[TrackerStore] = None,
         lock_store: Optional[LockStore] = None,
         action_endpoint: Optional[EndpointConfig] = None,
         fingerprint: Optional[Text] = None,
-        model_directory: Optional[Text] = None,
         model_server: Optional[EndpointConfig] = None,
         remote_storage: Optional[Text] = None,
-        path_to_model_archive: Optional[Text] = None,
+        graph_runner: Optional[GraphRunner] = None,
+        model_path: Optional[Text] = None,
+        model_id: Optional[Text] = None,
```

Comment:
How about passing in the entire model_metadata object instead of domain / `model_id` separately? In my opinion fine if `ModelMetadata` is widely exposed. Alternatively I'd not pass any of this and only retrieve this from the `Processor` 🤔

Reply:
I don't think we even need it on the agent anymore. Can just access through the processor.